provider/cluster_autoscaler_resource.go

```diff
@@ -0,0 +1,704 @@
+									},
+									"max": {
+										Type:     types.Int64Type,
+										Required: true,
+									},
+								}),
+								Validators: RangeValidators(),
+							},
+						}, tfsdk.ListNestedAttributesOptions{
+							MinItems: 1,
+						}),
+					},
+				}),
+			},
+			"scale_down": {
+				Description: "Configuration of scale down operation.",
+				Optional:    true,
+				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+					"enabled": {
+						Description: "Should cluster-autoscaler scale down the cluster.",
+						Type:        types.BoolType,
+						Optional:    true,
+					},
+					"unneeded_time": {
+						Description: "How long a node should be unneeded before it is eligible for scale down.",
+						Type:        types.StringType,
+						Optional:    true,
+					},
+					"utilization_threshold": {
+						Description: "Node utilization level, defined as sum of requested resources divided " +
+							"by capacity, below which a node can be considered for scale down.",
+						Type:       types.StringType,
+						Optional:   true,
+						Validators: FloatRangeValidators("utilization threshold validation", 0.0, 1.0),
+					},
+					"delay_after_add": {
+						Description: "How long after scale up that scale down evaluation resumes.",
+						Type:        types.StringType,
+						Optional:    true,
+						Validators:  DurationStringValidators("delay after add validation"),
+					},
+					"delay_after_delete": {
+						Description: "How long after node deletion that scale down evaluation resumes.",
+						Type:        types.StringType,
+						Optional:    true,
+						Validators:  DurationStringValidators("delay after delete validation"),
+					},
+					"delay_after_failure": {
+						Description: "How long after scale down failure that scale down evaluation resumes.",
+						Type:        types.StringType,
+						Optional:    true,
+						Validators:  DurationStringValidators("delay after failure validation"),
+					},
+				}),
+			},
+		},
+	}, nil
+}
+
+func RangeValidators() []tfsdk.AttributeValidator {
+	return []tfsdk.AttributeValidator{
+		&common.AttributeValidator{
+			Desc: "max must be greater or equal to min",
+			Validator: func(ctx context.Context, req tfsdk.ValidateAttributeRequest, resp *tfsdk.ValidateAttributeResponse) {
+				resourceRange := &AutoscalerResourceRange{}
+				diag := req.Config.GetAttribute(ctx, req.AttributePath, resourceRange)
+				if diag.HasError() {
+					// No attribute to validate
+					return
+				}
+
+				steps := []string{}
+				for _, step := range req.AttributePath.Steps() {
+					steps = append(steps, fmt.Sprintf("%s", step))
+				}
+				if resourceRange.Min.Value > resourceRange.Max.Value {
+					resp.Diagnostics.AddAttributeError(
+						req.AttributePath,
+						"Invalid resource range",
+						// fmt.Sprintf("In '%s.%s' attribute, max value must be greater or equal to min value", req.AttributePath.WithoutLastStep().LastStep(), req.AttributePath.LastStep()),
```

Comment:
Did you forget to change to the commented line, or forgot to delete? I assume delete. :)

Reply:
oops, forgot that
thanks!